contributing-docs/testing/unit_tests.rst

```diff
@@ -0,0 +1,1160 @@
+....................
+
+For the Non-DB tests, they are run once for each tested Python version with ``none`` database backend (which
+causes any database access to fail. Those tests are run with ``pytest-xdist`` plugin in parallel which
+means that we can efficiently utilised multi-processor machines (including ``self-hosted`` runners with
+8 CPUS we have to run the tests with maximum parallelism).
+
+It's usually straightforward to run those tests in local virtualenv because they do not require any
+setup or running database. They also run much faster than DB tests. You can run them with ``pytest`` command
+or with ``breeze`` that has all the dependencies needed to run all tests automatically installed. Of course
+you can also select just specific test or folder or module for the Pytest to collect/run tests from there,
+the example below shows how to run all tests, parallelising them with ``pytest-xdist``
+(by specifying ``tests`` folder):
+
+.. code-block:: bash
+
+    pytest tests --skip-db-tests -n auto
+
+
+The ``--skip-db-tests`` flag will only run tests that are not marked as DB tests.
+
+
+You can also run ``breeze`` command to run all the tests (they will run in a separate container,
+the selected python version and without access to any database). Adding ``--use-xdist`` flag will run all
+tests in parallel using ``pytest-xdist`` plugin.
+
+We have a dedicated, opinionated ``breeze testing non-db-tests`` command as well that runs non-DB tests
+(it is also used in CI to run the non-DB tests, where you do not have to specify extra flags for
+parallel running and you can run all the Non-DB tests
+(or just a subset of them with ``--parallel-test-types`` or ``--exclude-parallel-test-types``) in parallel:
+
+.. code-block:: bash
+
+    breeze testing non-db-tests
+
+You can pass ``--parallel-test-type`` list of test types to execute or ``--exclude--parallel-test-types``
+to exclude them from the default set:.
+
+.. code-block:: bash
+
+    breeze testing non-db-tests --parallel-test-types "Providers API CLI"
+
+
+.. code-block:: bash
+
+    breeze testing non-db-tests --exclude-parallel-test-types "Providers API CLI"
+
+You can also run the same commands via ``breeze testing tests`` - by adding the necessary flags manually:
+
+.. code-block:: bash
+
+    breeze testing tests --skip-db-tests --backend none --use-xdist
+
+Also you can enter interactive shell with ``breeze`` and run tests from there if you want to iterate
+with the tests. Source files in ``breeze`` are mounted as volumes so you can modify them locally and
+rerun in Breeze as you will (``-n auto`` will parallelize tests using ``pytest-xdist`` plugin):
+
+.. code-block:: bash
+
+    breeze shell --backend none --python 3.8
+    > pytest tests --skip-db-tests -n auto
+
+
+Airflow DB tests
+................
+
+Airflow DB tests require database to run. It can be any of the supported Airflow Databases and they can
+be run either using local virtualenv or Breeze
+
+
+
```

Comment:
DB tests are mentioned above as well - is this a redundancy-leftover or a history merge conflict?

Reply:
Yeah. I  rearranged it and moved things together and removed some duplication here.